test/specs/core.datasetController.tests.js

```diff
@@ -38,6 +38,22 @@ describe('Chart.DatasetController', function() {
 		});
 	});
 
+	it('should handle a frozen data object', function() {
+		function createChart() {
+			var data = Object.freeze([0, 1, 2, 3, 4, 5]);
+			acquireChart({
+				type: 'line',
+				data: {
+					datasets: [{
+						data: data
+					}]
+				}
+			});
+		}
+
+		expect(createChart).not.toThrow();
+	});
+
```

Comment:
We should add a test that goes through the [`unlistenArrayEvents`](/chartjs/Chart.js/pull/6060/files#diff-94f81287f896941164eceb600fe6dde0R225) path while the previous data was frozen.

Reply:
I updated the test I added to `destroy()` the chart at the end

Reply:
Did you include that change in this PR? I'm not sure I see what you're referring to

Reply:
@benmccann its in commit e9bd480

Reply:
ok. i misunderstood. thanks

Reply:
@etimberg after the chart is created, can we also replace the dataset `data` by a new frozen data array and call `update()` (this should check [this line](/chartjs/Chart.js/blob/master/src/core/core.datasetController.js#L225)).

Reply:
Done